vortex-datafusion/src/persistent/sink.rs

```diff
@@ -203,4 +209,61 @@ mod tests {
             2
         );
     }
+
+    #[tokio::test]
+    async fn write_to_object_storage() -> anyhow::Result<()> {
```

Comment:
This test requires running docker locally, which is a bit painful but I think its worth it here as it does fail without the `ObjectStoreWriter` fix.